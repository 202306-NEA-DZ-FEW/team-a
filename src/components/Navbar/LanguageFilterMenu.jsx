import { Transition, Menu } from "@headlessui/react";
import Link from "next/link";
import { Fragment } from "react";
import { TbLanguage } from "react-icons/tb";

function LanguageFilterMenu({ currentPath }) {
  //This is for classnames
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }
  return (
    <>
      <Menu as='div' className='text-left hidden lg:block mr-1'>
        <div>
          <Menu.Button className='w-full justify-center bg-white px-3 py-2 text-sm font-semibold text-gray-900 hover:bg-gray-50 btn-circle btn-ghost'>
            <TbLanguage className='w-6 h-6' />
          </Menu.Button>
        </div>

        <Transition
          as={Fragment}
          enter='transition ease-out duration-100'
          enterFrom='transform opacity-0 scale-95'
          enterTo='transform opacity-100 scale-100'
          leave='transition ease-in duration-75'
          leaveFrom='transform opacity-100 scale-100'
          leaveTo='transform opacity-0 scale-95'
        >
          <Menu.Items className='absolute right-0 z-10 mt-6 lg:mr-4 w-screen md:w-screen lg:w-56 origin-top-right rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none'>
            <div className='py-1'>
              <Menu.Item>
                {({ active }) => (
                  <Link
                    href={currentPath}
                    locale='en'
                    className={classNames(
                      active ? "bg-gray-100 text-gray-900" : "text-gray-700",
                      "block px-4 py-2 text-sm my-2"
                    )}
                  >
                    <span className='fi fi-gb rounded mr-2'></span> English
                  </Link>
                )}
              </Menu.Item>
              <Menu.Item>
                {({ active }) => (
                  <Link
                    href={currentPath}
                    locale='fr'
                    className={classNames(
                      active ? "bg-gray-100 text-gray-900" : "text-gray-700",
                      "block px-4 py-2 text-sm mb-2"
                    )}
                  >
                    <span className='fi fi-fr rounded mr-2'></span> Français
                  </Link>
                )}
              </Menu.Item>
              <Menu.Item>
                {({ active }) => (
                  <Link
                    href={currentPath}
                    locale='ar'
                    className={classNames(
                      active ? "bg-gray-100 text-gray-900" : "text-gray-700",
                      "block px-4 py-2 text-sm mb-2"
                    )}
                  >
                    <span className='fi fi-sa rounded mr-2'></span> العربية
                  </Link>
                )}
              </Menu.Item>
            </div>
          </Menu.Items>
        </Transition>
      </Menu>
    </>
  );
}

export default LanguageFilterMenu;
