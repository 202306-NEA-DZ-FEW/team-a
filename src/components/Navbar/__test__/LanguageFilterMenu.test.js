import renderer from "react-test-renderer";

import LanguageFilterMenu from "../LanguageFilterMenu";

it("it renders correctly", () => {
  const tree = renderer.create(<LanguageFilterMenu currentPath='/' />).toJSON();
  expect(tree).toMatchSnapshot();
});
