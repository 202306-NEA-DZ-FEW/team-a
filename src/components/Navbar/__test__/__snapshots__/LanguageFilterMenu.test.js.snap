// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="text-left hidden lg:block"
  data-headlessui-state=""
>
  <div>
    <button
      aria-expanded={false}
      aria-haspopup="menu"
      className="btn btn-sm outline-dashed transition-all duration-500 ease-in-out hover:outline-double outline-[2px] btn-circle btn-ghost"
      data-headlessui-state=""
      id="headlessui-menu-button-:r0:"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <svg
        className="w-6 h-6"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
          stroke="none"
        />
        <path
          d="M4 5h7"
        />
        <path
          d="M9 3v2c0 4.418 -2.239 8 -5 8"
        />
        <path
          d="M5 9c0 2.144 2.952 3.908 6.7 4"
        />
        <path
          d="M12 20l4 -9l4 9"
        />
        <path
          d="M19.1 18h-6.2"
        />
      </svg>
    </button>
  </div>
</div>
`;
